app/src/main/java/org/mozilla/fenix/settings/address/view/AddressEditorState.kt

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings.address.view
+
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.concept.storage.Address
+import org.mozilla.fenix.settings.address.AddressEditorFragment
+import org.mozilla.fenix.settings.address.AddressEditorFragment.Companion.availableCountries
+import org.mozilla.fenix.settings.address.AddressEditorFragment.Companion.defaultCountry
+
+/**
+ * The state for the [AddressEditorFragment].
+ *
+ * @property country The currently selected [AddressCountry].
+ */
+data class AddressEditorState(
+    val country: AddressCountry,
+)
+
+/**
+ * Value type representing properties determined by the country used in an [Address].
+ * This data is meant to mirror the data currently represented on desktop here:
+ * https://searchfox.org/mozilla-central/source/toolkit/components/formautofill/addressmetadata/addressReferences.js
+ */
+data class AddressCountry(
```

Comment:
will all of the state variable data types be prefaced with `Address`? Might seem silly with `AddressAddressLine2` or something.

Reply:
This value type isn't meant to represent a field of an address. If you have suggestions for clarity I'm totally open.

This was intended to represent the various formatting and other related data that we get from the keys in [this file on desktop](https://searchfox.org/mozilla-central/source/toolkit/components/formautofill/addressmetadata/addressReferences.js). There's things in there like available states/provinces, names of fields for the address form, required fields for validity, etc. 

The intent was to be able to expand this class as we brought more of those properties in. The loose plan is to create a static binding to that data that is generated during builds and expose it as an object in GV. This class may even be replaced completely by that object once it's available.

Reply:
Gotcha gotcha. Without that file, I was completely lost about the new data class haha. If this and other data classes are going to be used to read-in data in a similar manner to that file, what if it's something like `ValidAddressCountry`, or maybe just `Country`, or make it a child class of `AddressEditorState` and it could referenced like `AddressEditorState.Country`